### ISSUE TITLE
Resizing windows with yattac

### ISSUE THREAD

**Issue by LGUG2Z** [LGUG2Z/yatta#3]
Oh and also, is there supposed to be a way to resize windows? Or at least make it so manually resizing windows makes yatta aware of the new size? (Feel free to tell me to open another issue because i guess I'm getting off-topic 😁 )

_Originally posted by @eramdam in https://github.com/LGUG2Z/yatta/issues/2#issuecomment-803249946_

**Comment by LGUG2Z** [comment]
@eramdam This commit adds support for resizing BSP layouts via `yattac` commands: https://github.com/LGUG2Z/yatta/commit/d2f51d0c52fb7b63e9d0dd2269934643d303c93e

I'll be using this build for a few days to look for any bugs or regressions before I open up a PR to merge it into the master branch, it would be great if you also have the time to run a build from this commit locally and play around with the resizing operations.

**Comment by eramdam** [comment]
Will try to do that later this week! Would it be possible to make it so mouse-resizing updates the tree as well? (Just curious if that's even possible)

**Comment by eramdam** [comment]
I did notice some issues:

- If I trigger a resize too quickly on an Explorer window then the operation gets aborted/the focus gets lost
- It seems like you can resize a window to end up out of bounds in the current desktop
- The resized window seems to quickly flash when the resize operation happens


https://user-images.githubusercontent.com/1409924/112572508-84ae4380-8da7-11eb-95d2-74c531f23793.mp4

**Comment by LGUG2Z** [comment]
I think the first and third issue may be related to AHK, I modified my bindings to add an extra comma which has removed those issues for me: 

```; Increase the size of a given edge, Alt + Arrow Key
!Left::
Run, yattac.exe resize left increase, , Hide
return

!Right::
Run, yattac.exe resize right increase, , Hide
return

!Up::
Run, yattac.exe resize top increase, , Hide
return

!Down::
Run, yattac.exe resize bottom increase, , Hide
return

; Decrease the size of a given edge, Alt + Shift + Arrow Key
!+Left::
Run, yattac.exe resize left decrease, , Hide
return

!+Right::
Run, yattac.exe resize right decrease, , Hide
return

!+Up::
Run, yattac.exe resize top decrease, , Hide
return

!+Down::
Run, yattac.exe resize bottom decrease, , Hide
return
```

Can you share the timestamp of when the 2nd issue happens in the video? I have found and fixed two bugs related to explorer and out of display bounds resizing this morning, the first where explorer fires a FocusChange event before a Show event, which means that insertion order (current_window_idx + 1) is not respected, and the second where a window inserted before a resized window does not inherit the resize adjustments of the window it pushes further down the tree.

Details on some other bugs fixed are in the commit message: https://github.com/LGUG2Z/yatta/commit/e014a8d743422d3a0d17b678e98102ab1fce7069

**Comment by eramdam** [comment]
> Can you share the timestamp of when the 2nd issue happens in the video?

Argh I thought I captured it 😅 But basically if you have a window in the bottom left of the tree and increase its size on the left, then it can grow out of bounds. 

I'll try the newest commit later today/this week-end. Curious to know what the comma does in the AHK script (Im a AHK newbie 😛 )

**Comment by LGUG2Z** [comment]
Ah yeah, I see what you mean. I've added some checks to ensure that a resize command can't result in a layout plan that does not respect the bounds of a display: https://github.com/LGUG2Z/yatta/commit/8655bf29851bec7d64aa28ac4f9bed532f56f6de

**Comment by LGUG2Z** [comment]
Closed with #9.
